near/nep141-locker/src/storage.rs

```diff
@@ -0,0 +1,142 @@
+            V0((&'a TransferMessage, &'a AccountId)),
+        }
+
+        borsh::to_vec(&RefTransferMessageStorage::V0((message, account)))
+    }
+}
+
+#[near_bindgen]
+impl Contract {
+    #[payable]
+    pub fn storage_deposit(&mut self, account_id: Option<AccountId>) -> StorageBalance {
+        let account_id = account_id.unwrap_or_else(env::predecessor_account_id);
+        let amount = env::attached_deposit();
+        let storage = if let Some(mut storage) = self.accounts_balances.get(&account_id) {
+            storage.total = storage.total.saturating_add(amount);
+            storage.available = storage.available.saturating_add(amount);
+            storage
+        } else {
+            let min_required_storage_balance = self.required_balance_for_account();
+            let available = amount
+                .checked_sub(min_required_storage_balance)
+                .sdk_expect("The attached deposit is less than the minimum storage balance");
+            StorageBalance {
+                total: amount,
+                available,
+            }
+        };
+
+        self.accounts_balances.insert(&account_id, &storage);
+        storage
+    }
+
+    #[payable]
+    pub fn storage_withdraw(&mut self, amount: Option<NearToken>) -> StorageBalance {
+        assert_one_yocto();
+        let account_id = env::predecessor_account_id();
+        let mut storage = self
+            .storage_balance_of(&account_id)
+            .sdk_expect("The account is not registered");
+        let to_withdraw = amount.unwrap_or(storage.available);
+        storage.total = storage
+            .total
+            .checked_sub(to_withdraw)
+            .sdk_expect("The amount is greater than the total storage balance");
+        storage.available = storage
+            .available
+            .checked_sub(to_withdraw)
+            .sdk_expect("The amount is greater than the available storage balance");
+
+        self.accounts_balances.insert(&account_id, &storage);
+        storage
+    }
+
+    pub fn storage_balance_bounds(&self) -> StorageBalanceBounds {
+        StorageBalanceBounds {
+            min: self.required_balance_for_account(),
+            max: None,
+        }
+    }
+
+    pub fn storage_balance_of(&self, account_id: &AccountId) -> Option<StorageBalance> {
+        self.accounts_balances.get(&account_id)
+    }
+
+    pub fn required_balance_for_account(&self) -> NearToken {
+        let key_len = 64 + 4;
+        let value_len = borsh::to_vec(&StorageBalance {
+            total: NearToken::from_yoctonear(0),
+            available: NearToken::from_yoctonear(0),
+        })
+        .sdk_expect("ERR_BORSH")
+        .len() as u64;
+
+        env::storage_byte_cost()
+            .saturating_mul((Self::get_basic_storage() + key_len + value_len).into())
+    }
+
+    pub fn required_balance_for_init_transfer(
+        &self,
+        recipient: OmniAddress,
+        sender: OmniAddress,
+    ) -> NearToken {
```

Comment:
This is the most controversial part; we currently can't mock those values like for account id because:
1- we don't limit the String for OmniAddress::Near & OmniAddress::Sol
2- the OmniAddress::Near recipient could be with message `account:msg`